services/workflows-service/src/transaction/dtos/transaction-create.dto.ts

```diff
@@ -33,10 +34,14 @@ export class CounterpartyInfo {
   @ApiProperty({ required: false }) @IsString() @IsOptional() sortCode?: string;
   @ApiProperty({ required: false }) @IsString() @IsOptional() bankCountry?: string;
 
-  @IsOptional()
+  @ApiProperty({ type: OmitType(BusinessCreateDto, ['correlationId']) })
+  @ValidateIf(obj => !obj.endUserData)
+  @ValidateNested()
   businessData?: Omit<BusinessCreateDto, 'correlationId'>;
 
-  @IsOptional()
+  @ApiProperty({ type: OmitType(EndUserCreateDto, ['correlationId']) })
+  @ValidateIf(obj => !obj.businessData)
+  @ValidateNested()
   endUserData?: Omit<EndUserCreateDto, 'correlationId'>;
 }
```

Comment:
Can be implemented by ValidatorConstraintInterface which allow at least one field to be present
https://github.com/typestack/class-validator/issues/1581

```
import { ValidationArguments, ValidatorConstraint, ValidatorConstraintInterface } from 'class-validator';

@ValidatorConstraint({ name: 'atLeastOne', async: false })
export class AtLeastOneConstraint implements ValidatorConstraintInterface {
    validate(value: any, args: ValidationArguments) {
        const { object } = args;
        if (object.endUserData !== undefined) {
            object.atLeastOnePresent = object.endUserData;
        } else if (object.businessData !== undefined) {
            object.atLeastOnePresent = object.businessData;
        }
        return object.endUserData !== undefined || object.businessData !== undefined;
    }

    defaultMessage(args: ValidationArguments) {
        return 'At least one of endUserData or businessData must be present';
    }
}
```